svcdiscovery/plugin/svcdiscovery/sdcclient.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2020 SUSE
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svcdiscovery
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// SDCClient is the Service Discovery Controller Client used to make calls to
+// the service-discovery-controller job to discover internal routes to apps.
+type SDCClient struct {
+	httpClient *http.Client
+	sdcURLBase string
+}
+
+// Discover discovers internal app routes from the Service Discovery Controller
+// and returns the list of IPs from these discovered routes.
+func (sdcc *SDCClient) Discover(domainName string) ([]string, error) {
+	url := sdcc.sdcURLBase + domainName
```

Comment:
I think I'd be a lot happier if the URL is a proper [URL](https://pkg.go.dev/net/url?tab=doc#URL) instead of a giant string (mostly to have a better chance at escaping things properly). Especially since we actually do that, and just convert it to string too early…

Reply:
AFAICT, the way I'm doing it is way more efficient than dealing with a URL object here (or even fmt.Sprintf).
Since the `domainName` string comes from a DNS Question Name, can this ever get wrong?